src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingSearchHelpers.cs

```diff
@@ -138,31 +139,36 @@ private static void GetNameAndArity(string typeName, out string typeNameWithoutA
             var index = await TopLevelSyntaxTreeIndex.GetRequiredIndexAsync(document, cancellationToken).ConfigureAwait(false);
             foreach (var info in index.DeclaredSymbolInfos)
             {
-                // Fast check to see if this looks like a candidate.
-                if (info.TypeParameterCount == symbolArity &&
-                    info.Name == symbolName)
+                // Fast checks to see if this looks like a candidate.
+
+                // In non-strict mode, allow the type-parameter count to be mismatched.
+                if (query.Strict && info.TypeParameterCount != symbolArity)
+                    continue;
+
+                if (info.Name != symbolName)
+                    continue;
+
+                // If it's a method, the parameter count much match.
+                if (query.MethodName != null)
                 {
-                    // If it's a method, the parameter count much match.
-                    if (query.MethodName != null)
-                    {
-                        if (info.Kind is not (DeclaredSymbolInfoKind.Method or DeclaredSymbolInfoKind.ExtensionMethod) ||
-                            info.ParameterCount != query.MethodParameterCount)
-                        {
-                            continue;
-                        }
-                    }
-
-                    // Looking promising so far.  Check that the container matches what the caller needs.
-                    if (info.FullyQualifiedContainerName != container)
+                    if (info.Kind is not (DeclaredSymbolInfoKind.Method or DeclaredSymbolInfoKind.ExtensionMethod))
                         continue;
 
-                    // Unit testing needs to know the final span a location may be mapped to (e.g. with `#line` taken
-                    // into consideration).  So map that information here for them.
-                    tree ??= await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                    var mappedSpan = tree.GetMappedLineSpan(info.Span, cancellationToken);
-
-                    result.Add(new UnitTestingDocumentSpan(new DocumentSpan(document, info.Span), mappedSpan));
+                    // In non-strict mode, allow the parameter count to be mismatched.
+                    if (query.Strict && info.ParameterCount != query.MethodParameterCount)
+                        continue;
                 }
+
+                // Looking promising so far.  Check that the container matches what the caller needs.
+                if (info.FullyQualifiedContainerName != container)
```

Comment:
Same question about case sensitivity for VB

Reply:
yup.  i'll find a way to handle that.